src/graph/state.cc

```diff
@@ -142,6 +142,23 @@ namespace hpp {
         hppDout (error, "Edge not found");
         return 0;
       }
+
+      void State::addNumericalConstraint(const ImplicitPtr_t& numConstraint)
+      {
+        ComparisonTypes_t comp(numConstraint->comparisonType());
+        for(constraints::ComparisonType c : comp)
+        {
+          if (c == constraints::Equality)
+          {
+            throw std::logic_error("Failed to insert constraint \""
+	     + numConstraint->function().name()
+	     + "\" as a state constraint since it contains a comparison "
+             + "of type Equality");
+          }
+        }
+        GraphComponent::addNumericalConstraint(numConstraint->copy());
```

Comment:
Why do you do this copy ?

Reply:
To make sure that nobody will modify the comparison types after insertion.

Reply:
At the moment, this cannot be ensure anyway because you can access the non-const constraints so you can always modify it.

The user can always check that his graph is valid by using the graph validation tool. Among other, it checks that no parameterized constraint are in the states.

Reply:
All right. Then I remove the copy and push again.